# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/pam-no_nis.patch
# Copyright (C) 2004 - 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- Linux-PAM-0.99.6.3/modules/pam_access/pam_access.c.orig	2006-08-31 12:20:37.000000000 +0200
+++ Linux-PAM-0.99.6.3/modules/pam_access/pam_access.c	2006-09-28 07:22:35.000000000 +0200
@@ -41,7 +41,6 @@
 #include <errno.h>
 #include <ctype.h>
 #include <sys/utsname.h>
-#include <rpcsvc/ypclnt.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <sys/socket.h>
--- ./modules/pam_unix/yppasswd_xdr.c.orig	2005-01-19 22:34:53.000000000 -0300
+++ ./modules/pam_unix/yppasswd_xdr.c	2005-01-19 22:35:08.000000000 -0300
@@ -13,8 +13,6 @@
 #include <security/_pam_aconf.h>
 
 #include <rpc/rpc.h>
-#include <rpcsvc/yp_prot.h>
-#include <rpcsvc/ypclnt.h>
 #include "yppasswd.h"
 
 bool_t
--- Linux-PAM-0.99.2.1/modules/pam_unix/support.c.orig	2005-11-29 10:22:06.000000000 +0100
+++ Linux-PAM-0.99.2.1/modules/pam_unix/support.c	2006-02-01 20:59:06.000000000 +0100
@@ -21,7 +21,6 @@
 #include <ctype.h>
 #include <syslog.h>
 #include <sys/resource.h>
-#include <rpcsvc/ypclnt.h>
 
 #include <security/_pam_macros.h>
 #include <security/pam_modules.h>
@@ -257,25 +256,6 @@
 		}
 	}
 
-	if (!matched && nis) {
-		char *userinfo = NULL, *domain = NULL;
-		int len = 0, i;
-		len = yp_get_default_domain(&domain);
-		if (len == YPERR_SUCCESS) {
-			len = yp_bind(domain);
-		}
-		if (len == YPERR_SUCCESS) {
-			i = yp_match(domain, "passwd.byname", name,
-				     strlen(name), &userinfo, &len);
-			yp_unbind(domain);
-			if ((i == YPERR_SUCCESS) && ((size_t)len < sizeof(buf))) {
-				strncpy(buf, userinfo, sizeof(buf) - 1);
-				buf[sizeof(buf) - 1] = '\0';
-				matched = 1;
-			}
-		}
-	}
-
 	if (matched && (ret != NULL)) {
 		*ret = NULL;
 
--- ./modules/pam_unix/pam_unix_passwd.c.orig	2006-06-27 08:38:14.000000000 +0000
+++ ./modules/pam_unix/pam_unix_passwd.c	2006-11-15 13:57:41.000000000 +0000
@@ -54,8 +54,6 @@
 #include <sys/time.h>
 #include <sys/stat.h>
 #include <rpc/rpc.h>
-#include <rpcsvc/yp_prot.h>
-#include <rpcsvc/ypclnt.h>
 
 #include <signal.h>
 #include <errno.h>
@@ -187,6 +185,9 @@
 
 static char *getNISserver(pam_handle_t *pamh)
 {
+#if defined(__UCLIBC__)
+	return NULL;
+#else
 	char *master;
 	char *domainname;
 	int port, err;
@@ -213,6 +214,7 @@
 		return NULL;
 	}
 	return master;
+#endif
 }
 
 #ifdef WITH_SELINUX
@@ -757,6 +759,7 @@
 		goto done;
 	}
 
+#if !defined(__UCLIBC__)
 	if (on(UNIX_NIS, ctrl) && _unix_comesfromsource(pamh, forwho, 0, 1)) {
 	    if ((master=getNISserver(pamh)) != NULL) {
 		struct timeval timeout;
@@ -826,6 +829,7 @@
 		    retval = PAM_TRY_AGAIN;
 	    }
 	}
+#endif
 
 	if (_unix_comesfromsource(pamh, forwho, 1, 0)) {
 #ifdef USE_LCKPWDF
